sdk/src/Core/Amazon.Runtime/Credentials/FallbackCredentialsFactory.cs

```diff
@@ -20,12 +20,16 @@
 using System.IO;
 using System.Net;
 using System.Security;
+using System.Threading;
 
 namespace Amazon.Runtime
 {
     // Credentials fallback mechanism
     public static class FallbackCredentialsFactory
     {
+        // Lock to control caching credentials across multiple threads.
+        private static readonly object CredentialsLock = new object();
```

Comment:
Can you update this implementation to use `ReaderWriterLockSlim`? It is the recommended approach by dotnet for locking.